swing-lib/src/test/java/org/triplea/swing/JMenuBuilderTest.java

```diff
@@ -0,0 +1,62 @@
+package org.triplea.swing;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+import java.awt.event.KeyEvent;
+import javax.swing.JMenu;
+import javax.swing.JMenuItem;
+import org.junit.jupiter.api.Test;
+
+class JMenuBuilderTest {
+
+  private static final String TITLE = "title";
+  private static final String MENU_ITEM_TITLE = "menu-item-title";
+  private static final char MENU_MNEMONIC = 'M';
+
+  @Test
+  void verifyTitle() {
+    final JMenu menu = new JMenuBuilder(TITLE, MENU_MNEMONIC).build();
+    assertThat(menu.getText(), is(TITLE));
+  }
+
+  @Test
+  void verifyMnemonic() {
+    final JMenu menu = new JMenuBuilder(TITLE, 'M').build();
+    assertThat(menu.getMnemonic(), is(KeyEvent.VK_M));
+  }
+
+  @Test
+  void addMenuItem() {
+    final JMenu menu =
+        new JMenuBuilder(TITLE, MENU_MNEMONIC).addMenuItem(new JMenuItem(MENU_ITEM_TITLE)).build();
+
+    assertThat(menu.getItemCount(), is(1));
+    assertThat(menu.getItem(0).getText(), is(MENU_ITEM_TITLE));
+  }
+
+  @Test
+  void addMenuItemIf() {
+    JMenu menu =
+        new JMenuBuilder(TITLE, MENU_MNEMONIC)
+            .addMenuItemIf(false, MENU_ITEM_TITLE, 'a', () -> {})
+            .build();
+
+    assertThat(menu.getItemCount(), is(0));
+
+    menu =
+        new JMenuBuilder(TITLE, MENU_MNEMONIC)
+            .addMenuItemIf(true, MENU_ITEM_TITLE, 'a', () -> {})
+            .build();
+    assertThat(menu.getItemCount(), is(1));
+  }
+
+  @Test
+  void testAddMenuItem() {
+    final JMenu menu =
+        new JMenuBuilder(TITLE, MENU_MNEMONIC).addMenuItem(MENU_ITEM_TITLE, 'a', () -> {}).build();
+    assertThat(menu.getItemCount(), is(1));
+    assertThat(menu.getItem(0).getText(), is(MENU_ITEM_TITLE));
+    assertThat(menu.getItem(0).getMnemonic(), is(KeyEvent.VK_A));
+  }
```

Comment:
If swing were actually strict with its threading policy, this would cause exceptions to be thrown all over the place

Reply:
JavaFX is super strict when it comes to threading, everything needs to be done on the correct thread otherwise you'll get an exception.
At least this is consistent